web/src/components/molecules/Common/WorkspaceCreationModal/index.tsx

```diff
@@ -0,0 +1,59 @@
+import { Form, Input, Modal } from "antd";
+import React, { useCallback } from "react";
+
+export interface FormValues {
+  name: string;
+}
+
+export interface Props {
+  open?: boolean;
+  onClose?: (refetch?: boolean) => void;
+  onSubmit?: (values: FormValues) => Promise<void> | void;
+}
+
+const initialValues: FormValues = {
+  name: "",
+};
+
+const WorkspaceCreationModal: React.FC<Props> = ({
+  open,
+  onClose,
+  onSubmit,
+}) => {
+  const [form] = Form.useForm();
+
+  const handleSubmit = useCallback(() => {
+    form
+      .validateFields()
+      .then(async (values) => {
+        await onSubmit?.(values);
+        onClose?.(true);
+        form.resetFields();
+      })
+      .catch((info) => {
+        console.log("Validate Failed:", info);
+      });
+  }, [form, onClose, onSubmit]);
+
+  const handleClose = useCallback(() => {
+    onClose?.(true);
+  }, [onClose]);
```

Comment:
Same as another comment. Is this necessary? I am assuming you are trying to avoid unecessary re-renders, but if onClose is wrapped in a useCallback, isn't that enough?